boards/BMS/Src/App/states/App_AllStates.c

```diff
@@ -157,6 +158,16 @@ bool App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
         status = false;
         App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
     }
+    // Checks if the charger has thrown a fault, the disabling of the charger, etc is done with ChargeStateRunOnExit
+    else if (App_Charger_IsConnected(charger))
+    {
+        if (App_Charger_HasFaulted(charger))
+        {
+            status = false;
+            App_CanTx_SetPeriodicSignal_IS_CHARGING_ENABLED(can_tx, false);
+            App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+        }
```

Comment:
Doesn't look like fault state checks that charger faults are cleared before going back into init, so this would cause cycling from fault <-> init. Prevent fault -> init transition until charger faults are cleared.